mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,404 @@
+            ###probably due to specific boxy settings actually###
+            if 'bias-A' in raw_data.columns:
+                bias[detectors.index(i_detect),:] = raw_data['bias-' + i_detect].to_numpy()
+                
+            ###loop through data types###
+            for i_data in data_types:
+                ###loop through sources###
+                for i_source in sources:                    
+                    ###where to store our data###
+                    index_loc = detectors.index(i_detect)*source_num + (i_source-1)                    
+                    ###need to treat our filetypes differently###
+                    if filetype == 'non-parsed':
+                        
+                        ###filetype saves timepoints in groups###
+                        ###this should account for that###
+                        time_points = np.arange(i_source-1,int(record[-1])*source_num,source_num)
+                        
+                        ###determine which channel to look for###
+                        channel = i_detect + '-' + i_data
+                        
+                        ###save our data based on data type###
+                        if data_types.index(i_data) == 0:
+                            raw_ac[index_loc,:] = raw_data[channel][time_points].to_numpy()
+                        elif data_types.index(i_data) == 1:
+                            raw_dc[index_loc,:] = raw_data[channel][time_points].to_numpy()
+                        elif data_types.index(i_data) == 2:
+                            raw_ph[index_loc,:] = raw_data[channel][time_points].to_numpy()
+                    elif filetype == 'parsed':                    
+                        ###determine which channel to look for###
+                        channel = i_detect + '-' + i_data + str(i_source)
+                        
+                        ###save our data based on data type###
+                        if data_types.index(i_data) == 0:
+                            raw_ac[index_loc,:] = raw_data[channel].to_numpy()
+                        elif data_types.index(i_data) == 1:
+                            raw_dc[index_loc,:] = raw_data[channel].to_numpy()
+                        elif data_types.index(i_data) == 2:
+                            raw_ph[index_loc,:] = raw_data[channel].to_numpy()
+     
+        ###now combine our data types into a single array with the data###
+        data = np.append(raw_ac, np.append(raw_dc, raw_ph, axis=0),axis=0)
+
+
+        # Read triggers from event file
+        ###add our markers to the data array based on filetype###
+        if filetype == 'non-parsed':
+            if type(digaux) is list and digaux != []:
+                markers = digaux[np.arange(0,len(digaux),source_num)]
+            else:
+                markers = np.zeros(np.size(data,axis=1))
+        elif filetype == 'parsed':
+            markers = digaux  
+        data = np.vstack((data, markers))
```

Comment:
... change this line to:
```
data[:] = np.vstack((data_, markers))[:, start:stop]
```
I get an error that the `data_` you have constructed is the wrong size -- it's shape (385,6500) but the `start, stop` here require 104134 samples:
```
  File "/home/larsoner/python/mne-python/mne/io/boxy/boxy.py", line 403, in _read_segment_file
    data[:] = np.vstack((data_, markers))[:, start:stop]
ValueError: could not broadcast input array from shape (385,6500) into shape (385,104134)
```
Hopefully that allows you to continue debugging the reader...

Reply:
io/base.py ln 396 is where the _read_segment_file gets called with a int(start_file) and int(stop_file) which must be incorrect (probably based on the overall length of the file before ignoring header, footer? )

Reply:
self._first_samps[fi],
self._last_samps[fi]
don't seem to be correct

Reply:
Got this working and data plotting, thanks @larsoner, the input file has each source on its own row, but we had already rearranged them to their own columns but it was still expecting 16 times as many rows, fixed in latests push